From 18553f16cc6db4a415ec6efce05edca2990c45d3 Mon Sep 17 00:00:00 2001
From: hmgokhale <hmgokhale@maxlinear.com>
Date: Tue, 16 Jul 2024 20:54:52 +0530
Subject: [PATCH] WLANRTSYS-82731 Update the bss index after the reconf

Issue: hostapd segfault is seen after configuration of MLD using
UCI and issue wifi command

RCA: During wifi command while removing the sibling vap of MLD, it was
removing the non MLD vap which is not correct. This is due to the wrong
bss index used
During the reconf the VAP is added and deleted. During this process the
bss index for this VAP is not updated resulting in using the wrong index
while accessing the sibling hapd.

Fix: Update the bss index after the reconf

Signed-off-by: hmgokhale <hmgokhale@maxlinear.com>
---
 src/ap/hostapd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index 158b9df24..974e72a32 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -4386,6 +4386,7 @@ static int hostapd_reconf_add_bss(struct hostapd_iface *iface, struct hostapd_co
 	}
 
 	idx = iface->num_bss - 1;
+	iface->bss[idx]->conf->bss_idx = idx;
 	hostapd_flush_old_stations(iface->bss[idx], WLAN_REASON_PREV_AUTH_NOT_VALID);
 #ifdef CONFIG_IEEE80211BE_MXL_MLO
 	if(mxl_mld_setup_interfaces(iface->interfaces))
-- 
2.43.0

